src/main/java/com/uber/profiling/profilers/ProcessInfoBase.java

```diff
@@ -16,14 +16,15 @@
 
 package com.uber.profiling.profilers;
 
+import com.uber.profiling.Profiler;
 import com.uber.profiling.util.NetworkUtils;
 import com.uber.profiling.util.ProcFileUtils;
 import com.uber.profiling.util.ProcessUtils;
 import com.uber.profiling.util.SparkUtils;
 
 import java.util.UUID;
 
-public class ProcessInfoBase {
+public abstract class ProcessInfoBase extends Profiler {
```

Comment:
The semantic for class inheritance is establishing "is-a" relationship, see [here](https://en.wikipedia.org/wiki/Inheritance_(object-oriented_programming)). ProcessInfoBase should not be a Profiler.

Reply:
as part of this change, all classes from `com.uber.profiling.profilers` are now profilers.

Agree that `ProcessInfoBase extends Profiler` seems forced naming wise, but if you look for example at `ProcessInfoProfiler extends ProcessInfoBase` there is a profiler eventually extending base `Profiler`

So my suggestion would be to `ProcessInfoBase` to `ProcessInfoBaseProfiler`.
Would it make sense then?

Reply:
Yeah, the old code "ProcessInfoProfiler extends ProcessInfoBase" looks not good as well. Let me do some refactor there.

Reply:
I updated the code to rename ProcessInfoBase to ProfilerBase, which makes more sense now.

Reply:
ok - rebased on top of master